test/unit/controllers/consolidations/AssociateDucrSummaryControllerSpec.scala

```diff
@@ -73,131 +61,101 @@ class AssociateDucrSummaryControllerSpec extends ControllerSpec with MockSubmiss
-        inOrder
-          .verify(mockCustomsCacheService)
-          .fetchAndGetEntry[AssociateDucr](any(), meq(AssociateDucr.formId))(any(), any(), any())
-        inOrder
-          .verify(mockSubmissionService)
-          .submitDucrAssociation(meq(MucrOptions("MUCR")), meq(AssociateDucr("DUCR")))(any(), any())
-        inOrder.verify(mockCustomsCacheService).remove(anyString)(any(), any())
+        assertThrows[IncompleteApplication] {
+          await(controller.displayPage()(getRequest()))
+        }
       }
 
-      "Redirect to next page" in {
-
-        when(mockSubmissionService.submitDucrAssociation(any(), any())(any(), any()))
-          .thenReturn(Future.successful(ACCEPTED))
-        when(mockCustomsCacheService.remove(anyString())(any(), any()))
-          .thenReturn(Future.successful(mock[HttpResponse]))
+      "Mucr Options is missing during submitting page" in {
 
-        val result = controller.submit()(postRequest(Json.obj()))
+        withCaching(MucrOptions.formId, None)
 
-        redirectLocation(result) mustBe Some(routes.AssociateDucrConfirmationController.displayPage().url)
+        assertThrows[IncompleteApplication] {
+          await(controller.submit()(postRequest(Json.obj())))
+        }
       }
 
-      "add MUCR to Flash" in {
-
-        when(mockSubmissionService.submitDucrAssociation(any(), any())(any(), any()))
-          .thenReturn(Future.successful(ACCEPTED))
-        when(mockCustomsCacheService.remove(anyString())(any(), any()))
-          .thenReturn(Future.successful(mock[HttpResponse]))
+      "Associate Ducr is missing during submitting page" in {
 
-        val result = controller.submit()(postRequest(Json.obj()))
+        withCaching(MucrOptions.formId, Some(MucrOptions("MUCR")))
+        withCaching(AssociateDucr.formId, None)
 
-        flash(result).get(FlashKeys.MUCR) mustBe Some("MUCR")
+        assertThrows[IncompleteApplication] {
+          await(controller.submit()(postRequest(Json.obj())))
+        }
       }
-
     }
 
-    "SubmissionService returns status other than Accepted" should {
+    "return 303 (SEE_OTHER)" when {
 
-      "return InternalServerError code" in {
+      "submission service returned ACCEPTED and all mandatory data is in cache" in {
 
-        when(mockSubmissionService.submitDucrAssociation(any(), any())(any(), any()))
-          .thenReturn(Future.successful(BAD_REQUEST))
-        when(mockCustomsCacheService.remove(anyString())(any(), any()))
-          .thenReturn(Future.successful(mock[HttpResponse]))
+        withCaching(MucrOptions.formId, Some(MucrOptions("MUCR")))
+        withCaching(AssociateDucr.formId, Some(AssociateDucr("DUCR")))
+        mockCustomsCacheServiceClearedSuccessfully()
+        mockDucrAssociation()
 
         val result = controller.submit()(postRequest(Json.obj()))
 
-        status(result) mustBe INTERNAL_SERVER_ERROR
+        status(result) mustBe SEE_OTHER
```

Comment:
Where do we assert that the right param is in the `Flash`?

Reply:
`flash(result).get("some key") mustBe "xyz"`

Reply:
Fixed thanks